src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -125,13 +125,16 @@ case class CsvRelation protected[spark] (
           case aiob: ArrayIndexOutOfBoundsException if permissive =>
             (index until schemaFields.length).foreach(ind => rowArray(ind) = null)
             Some(Row.fromSeq(rowArray))
-          case _: java.lang.NumberFormatException |
-               _: IllegalArgumentException if dropMalformed =>
-            logger.warn("Number format exception. " +
+          case e: java.lang.NumberFormatException if dropMalformed =>
+            logger.warn("Number format exception  (" + e + ")." +
```

Comment:
In addition, you can write `s"Number format exception  (${a.getMessage})."` instead of concatenating with `+` which is not used generally in this library and Spark codes as well.